api/objects.proto

```diff
@@ -249,6 +249,14 @@ message Cluster {
 	// If the key is empty, the node will be unlocked (will not require a key
 	// to start up from a shut down state).
 	repeated EncryptionKey unlock_keys = 9;
+	
+	// RoutingMeshOff denotes whether the routing mesh is currently disabled.
+	// By default the routing mesh is enabled. As result of user removing the
+	// ingress network, this flag will be set to true. Cluster needs to carry
+	// this information, so that on reboot or leader re-election, new leader
+	// will not automatically create the routing mesh if it cannot find one
+	// in store. 
+	bool routingMeshOff = 10;
```

Comment:
I think it may be possible to avoid keeping this state. Right now we create the default cluster object here:

https://github.com/docker/swarmkit/blob/a5eb9c0c8c82c544819f3811f9bfba44c1546998/manager/manager.go#L882

What if we created the ingress network at the same time, but only if the `CreateCluster` call succeeded? This would ensure that a swarm cluster always gets an ingress network by default, but if it is deleted later on, it would not be recreated automatically?

I'm not sure if this would be sort of hacky. It seems like it would simplify things a bit, but I don't have a strong opinion.